internal/factsengine/gatherers/corosynccmapctl.go

```diff
@@ -0,0 +1,69 @@
+package gatherers
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/factsengine/entities"
+	"github.com/trento-project/agent/internal/utils"
+)
+
+const (
+	CorosyncCmapCtlGathererName = "corosync-cmapctl"
+)
+
+// nolint:gochecknoglobals
+var (
+	CorosyncCmapCtlValueNotFound = entities.FactGatheringError{
+		Type:    "corosync-cmapctl-value-not-found",
+		Message: "requested value not found in corosync-cmapctl output",
+	}
+
+	CorosyncCmapCtlCommandError = entities.FactGatheringError{
+		Type:    "corosync-cmapctl-command-error",
+		Message: "error while executing corosynccmap-ctl",
+	}
+)
+
+type CorosyncCmapctlGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultCorosyncCmapctlGatherer() *CorosyncCmapctlGatherer {
+	return NewCorosyncCmapctlGatherer(utils.Executor{})
+}
+
+func NewCorosyncCmapctlGatherer(executor utils.CommandExecutor) *CorosyncCmapctlGatherer {
+	return &CorosyncCmapctlGatherer{
+		executor: executor,
+	}
+}
+
+func (s *CorosyncCmapctlGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s facts gathering process", CorosyncCmapCtlGathererName)
+
+	corosyncCmapctl, err := s.executor.Exec(
+		"corosync-cmapctl", "-b")
+	if err != nil {
+		return nil, CorosyncCmapCtlCommandError.Wrap(err.Error())
+	}
+
+	corosyncCmapctlMap := utils.FindMatches(`(?m)^(\S*)\s\(\S*\)\s=\s(.*)$`, corosyncCmapctl)
+
+	for _, factReq := range factsRequests {
+		var fact entities.Fact
+
+		if value, ok := corosyncCmapctlMap[factReq.Argument]; ok {
+			fact = entities.NewFactGatheredWithRequest(factReq, &entities.FactValueString{Value: fmt.Sprint(value)})
```

Comment:
Use [ParseStringToFactValue](https://github.com/trento-project/agent/blob/main/internal/factsengine/entities/fact_value.go#L91).
We have plenty of integers here

Reply:
We might want to consider also lists/maps.
For instance we need to determine that _Corosync is running with at least 2 rings_, meaning that in the command output we'd find something like the following (if I understood Check 32CFC6)
```shell
$ corosync-cmapctl -b
...
totem.interface.0.bindnetaddr (str) = 10.80.1.12
...
totem.interface.1.bindnetaddr (str) = 10.80.1.13
...
```
What should we pass as an argument to this gatherer to get the rings and use them in the Spec via `facts.running_corosync_rings.len() == 2`?

This can be very well deferred if too much for this PR, anyway this gatherer already allows to deliver 7 Checks :rocket:

Reply:
@nelsonkopliku That's a really fair point.
I didn't consider it myself.
From my side, if it is easy to implement. We could/should do it. It would give the gatherer much more flexibility. Otherwise we are limited to gather ending values.
@rtorrero maybe we can check how easy it is?
I'm not against merging what we have now, but we should track the debt

Reply:
@CDimonaco maybe you have some idea how we could do this transformation? from dot separated strings to a map